shared/middleware/files_api.rb

```diff
@@ -113,12 +119,53 @@ def record_event(quota_event_type, quota_type, encrypted_channel_id)
   # Read a file. Optionally get a specific version instead of the most recent.
   #
   get %r{/v3/(animations|assets|sources|files)/([^/]+)/([^/]+)$} do |endpoint, encrypted_channel_id, filename|
+    get_file(endpoint, encrypted_channel_id, filename)
+  end
+
+  #
+  # GET /<channel-id>/<filename>?version=<version-id>
+  #
+  # Read a file. Optionally get a specific version instead of the most recent.
+  # Only from codeprojects.org domain
+  #
+  get %r{/([^/]+)/([^/]+)$}, { code_projects_domain: true } do |encrypted_channel_id, filename|
+    pass unless valid_encrypted_channel_id(encrypted_channel_id)
+
+    get_file('files', encrypted_channel_id, filename, true)
+  end
+
+  #
+  # GET /<channel-id>
+  #
+  # Redirect to /<channel-id>/
+  # Only from codeprojects.org domain
+  #
+  get %r{/([^/]+)$}, { code_projects_domain: true } do |encrypted_channel_id|
+    pass unless valid_encrypted_channel_id(encrypted_channel_id)
+
+    redirect "#{request.path_info}/"
+  end
+
+  #
+  # GET /<channel-id>/
+  #
+  # Serve index.html for this project.
+  # Only from codeprojects.org domain
+  #
+  get %r{/([^/]+)/$}, { code_projects_domain: true } do |encrypted_channel_id|
+    pass unless valid_encrypted_channel_id(encrypted_channel_id)
+
+    get_file('files', encrypted_channel_id, 'index.html', true)
```

Comment:
Looks like this will be an easy extension point if/when we want to also serve app lab or game lab projects from this domain.  Great!